### PR TITLE
Static resources additional guidance

### PR DESCRIPTION
**Pull request by roastlechon**

### CODE REVIEW

File: docs/functional-components/static-assets.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+## Recommended packages
+
+* [express-static][]: `express-static` is part of the Express.js package that allows developers to expose static middlewares.
+
+## Guidance
+
+For serving static resources on the application, we recommend using [express-static][] middleware as it has been widely used and tested in production.
+
+When using express.static() method, we can serve static resources directly by specifying the folder name where we have stored our static resources.
```
Comment (wtrocki) [inline docs/functional-components/static-assets.md:11]:
```suggestion
When using `express.static()` method, we can serve static resources directly by specifying the folder name where we have stored our static resources.
```

File: docs/functional-components/static-assets.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Documentation for the middleware can be found [here][express-static]
+
+## Caching
+
+The static middleware allows caching of static resources via exposed [caching-headers][].
+
+Origin servers communicate caching instructions via the header `Cache-Control`. The values of `Cache-Control` are called directives. Example directives include: `max-age`, `no-store, no-cache, must-revalidate`, `public, private`, etc.
+
+Freshness control of the resource happens in cache and is based on time. The validation happens on origin server is based on time and identifiers (ETags).
```
Comment (wtrocki) [inline docs/functional-components/static-assets.md:21]:
```suggestion
Freshness control of the resource happens in cache and is based on time. The validation that happens on origin server is based on time and identifiers (ETags).
```

**Comment by wtrocki** [comment]
World class guide. Very clear and concise.
Amazing work!

File: docs/functional-components/static-assets.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+## Guidance
+
+For serving static resources on the application, we recommend using [express-static][] middleware as it has been widely used and tested in production.
+
+When using `express.static()` method, we can serve static resources directly by specifying the folder name where we have stored our static resources.
+
+Documentation for the middleware can be found [here][express-static]
+
+## Caching
```
Comment (mhdawson) [inline docs/functional-components/static-assets.md:15]:
```suggestion
### Caching
```

**Review by mhdawson — APPROVED** [review]
Other than a small formatting example LGTM

**Comment by lholmquist** [comment]
Looks good.  If we are recommending the `express-static` module, i think we should probably add it to the list , https://github.com/nodeshift/nodejs-reference-architecture/blob/master/modules-list.json

**Comment by roastlechon** [comment]
> Looks good. If we are recommending the `express-static` module, i think we should probably add it to the list , https://github.com/nodeshift/nodejs-reference-architecture/blob/master/modules-list.json

I fixed the references from express-static to express.static since it is not an npm module but built into express itself.

File: docs/functional-components/static-assets.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+When using `express.static()` method, we can serve static resources directly by specifying the folder name where we have stored our static resources.
+
+Documentation for the middleware can be found [here][express.static]
+
+### Caching
+
+The static middleware allows caching of static resources via exposed [caching-headers][].
+
+Origin servers communicate caching instructions via the header `Cache-Control`. The values of `Cache-Control` are called directives. Example directives include: `max-age`, `no-store, no-cache, must-revalidate`, `public, private`, etc.
```
Comment (roastlechon) [inline docs/functional-components/static-assets.md:19]: Expand directives to particular use cases:

Eg. depending on content, use a particular directive

Comment (mhdawson) [reply docs/functional-components/static-assets.md:19]: I think you also indicated you'd update to include 

* static content only (ie no need to run JS) -> just use nginx
* static content/with JS -> pretty much what we have, bundle together, rely on CDN to cache the static content effectively
* static content that is uploaded -> separate out

**Comment by mhdawson** [comment]
@nodeshift/reference-architecture this looks ready to land to me. Please review/approve or add any additional comments. I'll land early next week if I don't see any additional comments.

**Comment by wtrocki** [comment]
PR needs to be rebased. Reviewed from the content point view.

**Review by wtrocki — APPROVED** [review]

**Comment by roastlechon** [comment]
> PR needs to be rebased. Reviewed from the content point view.

Can you explain what needs to be rebased? The PR shows that its mergeable as is without needing to be rebased.

**Comment by wtrocki** [comment]
My bad. I had rebase and merge enabled.

**Review by lholmquist — APPROVED** [review]

**Pull request merged**